mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -3652,13 +3672,24 @@ func expandMonitoringConfig(configured interface{}) *container.MonitoringConfig
 	if len(l) == 0 || l[0] == nil {
 		return nil
 	}
-
+	mc := &container.MonitoringConfig{}
 	config := l[0].(map[string]interface{})
-	return &container.MonitoringConfig{
-		ComponentConfig: &container.MonitoringComponentConfig{
-			EnableComponents: convertStringArr(config["enable_components"].([]interface{})),
-		},
+<%# In version == 'ga' enable_components will always be specified. %>
+	if v, ok := config["enable_components"]; ok && len(v.([]interface{})) > 0 {
+		enable_components := v.([]interface{})[0].([]interface{})
```

Comment:
This seems to be causing a panic:

```
panic: interface conversion: interface {} is string, not []interface {}

goroutine 2274 [running]:
github.com/hashicorp/terraform-provider-google-beta/google-beta.expandMonitoringConfig(0x33dac00, 0xc002037ea8, 0x11)
	/go/src/github.com/hashicorp/terraform-provider-google-beta/google-beta/resource_container_cluster.go:3592 +0x345
```

Downstream diff link: https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5812-old..auto-pr-5812#diff-c173cfcd7737c1477299c920117b36a0aa5570f842b93f5e07ccb383b4157947R3592

It looks like the logic changed slightly compared to the old version? I'm guessing that restoring the old conversion will fix this.

```suggestion
		enable_components := v.([]interface{})
```

Reply:
@mbarrien - are you still working on this feature?

I wanted to call out that the feature has been promoted from beta to [GA](https://pkg.go.dev/google.golang.org/api/container/v1#MonitoringConfig) on the GKE API. So you shouldn't need to guard the API around `beta` anymore.